src/content/en/fundamentals/engage-and-retain/push-notifications/good-notification.markdown

```diff
@@ -0,0 +1,243 @@
+supported on all platforms, so you may need to fetch the critical information
+before displaying it.
+
+Until recently only mobile apps could do this. With service workers you can
+store a notification until a user wants it. When the user clicks it, the status
+of the network is irrelevant.
+
+{% highlight javascript %}
+self.addEventListener('push', event => {
+  var dataPromise;
+  if (data in event) {
+    dataPromise = Promise.resolve(event.data.json());
+  } else {
+    dataPromise = fetch('notification/end/point/data.json')
+      .then(response => {
+        return response.json();
+      });
+  }
+
+  event.waitUntil(
+    dataPromise
+    .then(msgData => {
+      // Now tell the user.
+      return self.registration.showNotification(data.title, {
+        // Whether you show data and how much you show depends on
+        // content of the data itself.
+        body: event.data.body,
+        icon: 'images/icon.png'
+      });
+    })
+  );
+}); 
+{% endhighlight %}
+
+### Use vibration judiciously {#vibrate-judiciously}
+
+Vibration may seem like an odd thing to list under timely. In fact, it's closely
+related and there are several issues.
+
+First, vibration may seem like the ideal way to make users aware of new
+notifications. But not all users have vibrations turned on and some devices
+can't vibrate. Consequently, any urgency that you intend to communicate with
+vibration can be lost.
+
+Second, making every notification vibrate can create a false sense of urgency.
+If users are being nagged by notifications that aren't as important as they
+seem, they may turn off notifications entirely.
+
+In short, let the user decide how to use vibrations. Give them the ability to
+select which notifications use vibrations or whether to use them at all. If you
+have different categories of notifications, you could even let them select
+different vibration patterns.
+
+Finally, remember that to vibrate a mobile device has to run a motor and motors
+use more power than an on-screen notification.
+
+## Precise {#precise}
+
+A precise notification is one that has specific information that can be acted on
+immediately. Consider the image from the anatomy lesson again.
+
+<img src="images/joe-asked.png" width="300" height="109" />
+
+It tells you:
+
+* Who sent the message.
+* An image so that you don't confuse the sender with any old average Joe.
+* Since the notification is a question, you can turn him down with as little
+  effort as is humanly possible.
+
+### Offer enough information that the user doesn't need to visit your site {#offer-enough}
+
+This may not be appropriate in every case, but if the information is simple
+enough to convey in a small space, don't make users open your web site to read
+it. For example, if you wanted to notify one user of acknowledgment by another,
+then don't show a message that says, "New notification". Show one that says,
+"Pete said, 'no'."
+
+Good:
```

Comment:
For good/bad images, use the following:
```
<div class="mdl-grid">
  <figure class="mdl-cell mdl-cell--6-col">
    <img src="images/hpnav-cta-good.png">
    <figcaption class="wf-figcaption-good"><b>Do:</b> make all of your users’ most common tasks easily available.</figcaption>
  </figure>
  <figure class="mdl-cell mdl-cell--6-col">
    <img src="images/hpnav-cta-bad.png">
    <figcaption class="wf-figcaption-bad"><b>Don’t:</b> waste precious above-the-fold space with vague calls-to-action like “learn more”.</figcaption>
  </figure>
</div>
```
It will end up looking like https://developers.google.com/web/fundamentals/getting-started/principles/site-and-page-navigation?hl=en